### PR TITLE
petric: get_data fixes

### LINKED ISSUE #31: `import petric` reads all data and other comments
When experimenting on one data-set, it is convenient to `import petrid` to get `petric.get_data()`. However, currently the import actually reads all (3) data-sets. That takes some time and a lot of memory (and will just become worse). This is due to
https://github.com/SyneRBI/PETRIC/blob/58750541908e4ace9d47e4aed731c709628574ca/petric.py#L171-L181
I suggest to convert `data_metric_pairs` a list of pairs of `srcdir`, `outdir` and `metrics`. This will also have the effect that the redirection
https://github.com/SyneRBI/PETRIC/blob/58750541908e4ace9d47e4aed731c709628574ca/petric.py#L155-L156
will write files in the relevant outdir when `get_data` is called (currently, all redirections will be in the folder for the last `get_data` call).
(By the way, the redirection filenames should not use the `BSREM` prefix.)

Also, I see no reason to have
https://github.com/SyneRBI/PETRIC/blob/58750541908e4ace9d47e4aed731c709628574ca/petric.py#L182-L183

### PR DESCRIPTION
**Pull request by KrisThielemans**
No longer read data upfront

Fixes #31

### CODE REVIEW

File: petric.py [hunk 1]
```diff
@@ -169,26 +169,25 @@ def get_data(srcdir=".", outdir=OUTDIR, sirf_verbosity=0):
...
-        (get_data(srcdir=SRCDIR / "Siemens_Vision600_thorax",
-                  outdir=OUTDIR / "Vision600_thorax"), [MetricsWithTimeout(outdir=OUTDIR / "Vision600_thorax")])]
+        (SRCDIR / "Siemens_Vision600_thorax", OUTDIR / "Vision600_thorax",
+         [MetricsWithTimeout(outdir=OUTDIR / "Vision600_thorax")])]
 else:
     log.warning("Source directory does not exist: %s", SRCDIR)
-    data_metrics_pairs = [(None, [])]
-# first dataset
-data, metrics = data_metrics_pairs[0]
+    data_dirs_metric = [(None, None, [])]
```
Comment (casperdcl) [inline petric.py:181]:
```suggestion
    data_dirs_metrics = [(None, None, [])]
```

File: petric.py [hunk 2]
```diff
@@ -169,26 +169,25 @@ def get_data(srcdir=".", outdir=OUTDIR, sirf_verbosity=0):
...
 
 if __name__ == "__main__":
```
Comment (casperdcl) [inline petric.py:183]:
```suggestion
if __name__ != "__main__":
    srcdir, outdir, metrics = data_dirs_metrics[0]
    data = get_data(srcdir=srcdir, outdir=outdir)
    metrics[0].reset()
else:
```

**Review by casperdcl — CHANGES_REQUESTED** [review]

File: petric.py [hunk 3]
```diff
@@ -169,26 +169,25 @@ def get_data(srcdir=".", outdir=OUTDIR, sirf_verbosity=0):
...
     from docopt import docopt
     args = docopt(__doc__)
     logging.basicConfig(level=getattr(logging, args["--log"].upper()))
     from main import Submission, submission_callbacks
     assert issubclass(Submission, Algorithm)
-    for data, metrics in data_metrics_pairs:
+    for srcdir, outdir, metrics in data_dirs_metric:
```
Comment (casperdcl) [inline petric.py:189]:
```suggestion
    for srcdir, outdir, metrics in data_dirs_metrics:
```

**Review by casperdcl — CHANGES_REQUESTED** [review]

File: petric.py (hunk 2 above)
Comment (KrisThielemans) [reply petric.py:183]: why do we need this?

Comment (casperdcl) [reply petric.py:183]: It's documented in the readme & main.py files (to keep docs simple)

**Comment by KrisThielemans** [comment]
As the `full` Action actually doesn't run, I guess we don't really know if this works or not. @casperdcl, save to merge after the `check` succeeds?

**Pull request merged**